### PR TITLE
Disable ent on 32 bit

### PR DESCRIPTION
**Pull request by jeffmendoza**
# Description of the PR

Disable ent on 32 bit

supersedes #1201

# PR Checklist

- [ ] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

**Review by pxp928 — APPROVED** [review]

File: cmd/guacgql/cmd/server.go [hunk 1]
```diff
@@ -264,3 +204,45 @@ func getAWSRequestSigner() (*v4.Signer, error) {
...
+	return neo4j.GetBackend(args)
+}
+
+func getInMem(ctx context.Context) (backends.Backend, error) {
+	args := &inmem.DemoCredentials{}
+	return inmem.GetBackend(args)
+}
+
+func getNeptune(ctx context.Context) (backends.Backend, error) {
+	// TODO: rename the neo4j config to something more generic since it would be used by Neptune as well.
```
Comment (mihaimaruseac) [inline cmd/guacgql/cmd/server.go:233]:
```suggestion
```

We no longer need this TODO

**Review by mihaimaruseac — APPROVED** [review]
I like this refactoring! :100:

File: cmd/guacgql/cmd/server.go [hunk 2]
```diff
@@ -51,6 +53,20 @@ const (
...
+
+func init() {
+	if getBackend == nil {
+		getBackend = make(map[string]backendFunc)
+	}
+	getBackend[arango] = getArango
+	getBackend[neo4js] = getNeo4j
+	getBackend[inmems] = getInMem
+	getBackend[neptune] = getNeptune
+}
```
Comment (lumjjb) [inline cmd/guacgql/cmd/server.go:68]: can we maybe move this to the backends pakage that will return this map and instead do that there? I think that may be another appropriate place to put it and we can dedup the code if other CLIs or unit tests need to do this as well.

File: cmd/guacgql/cmd/ent.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+package cmd
+
+import (
+	"context"
+
+	"github.com/guacsec/guac/pkg/assembler/backends"
+	entbackend "github.com/guacsec/guac/pkg/assembler/backends/ent/backend"
+)
+
+func init() {
```
Comment (lumjjb) [inline cmd/guacgql/cmd/ent.go:27]: ahh i wasn't thinking of it this way, but this works as well - i actually thought of doing this in the ent backend package itself.

**Review by lumjjb — APPROVED** [review]
LGTM, some comments

File: cmd/guacgql/cmd/server.go (hunk 2 above)
Comment (mihaimaruseac) [reply cmd/guacgql/cmd/server.go:68]: +1 to this, if possible.

File: cmd/guacgql/cmd/ent.go (hunk 3 above)
Comment (jeffmendoza) [reply cmd/guacgql/cmd/ent.go:27]: Yeah, even if the backend is ignored, the cmd can't have any code that pulls it in, so just do it here.

File: cmd/guacgql/cmd/server.go (hunk 2 above)
Comment (jeffmendoza) [reply cmd/guacgql/cmd/server.go:63]: Updated, ptal. Still need to setup `entbackend.BackendOptions` from cli flags in a 64-bit-only file.

File: pkg/assembler/backends/inmem/certifyLegal_test.go [hunk 4]
```diff
@@ -485,7 +485,7 @@ func TestLegal(t *testing.T) {
 	ctx := context.Background()
 	for _, test := range tests {
 		t.Run(test.Name, func(t *testing.T) {
-			b, err := inmem.GetBackend(nil)
+			b, err := backends.Get("inmem", nil, nil)
```
Comment (pxp928) [inline pkg/assembler/backends/inmem/certifyLegal_test.go:488]: Looks good! Just need to update the arango tests with these also

**Review by pxp928 — APPROVED** [review]

File: pkg/assembler/backends/inmem/certifyLegal_test.go (hunk 4 above)
Comment (jeffmendoza) [reply pkg/assembler/backends/inmem/certifyLegal_test.go:488]: done

File: cmd/guacgql/cmd/server.go (hunk 2 above)
Comment (lumjjb) [reply cmd/guacgql/cmd/server.go:64]: I was also thinking to do the registration within the package, kind of like we do for the parsers, and other pluggable things: 

https://github.com/guacsec/guac/blob/main/pkg/ingestor/parser/parser.go#L41-L51

Comment (jeffmendoza) [reply cmd/guacgql/cmd/server.go:64]: Yes, I've added that as you requested in the third commit on this PR:

https://github.com/guacsec/guac/pull/1226/commits/144d04501776e54e6b06d70d4f01c42ab1cef0ff#diff-1aba4420a9017b9d486c0fab38a08b9eea2e9c22e2402039d7726f26b1ccf544R164-R167
https://github.com/guacsec/guac/pull/1226/commits/144d04501776e54e6b06d70d4f01c42ab1cef0ff#diff-84928fb53202e5f8faf758692f063b2919686aec75b8d6d22381bc6e612a3e81R22-R24
And then the call uses `backend.Get` here https://github.com/guacsec/guac/pull/1226/commits/144d04501776e54e6b06d70d4f01c42ab1cef0ff#diff-c5774aa38759e9db3bd03f13cbd58be3e05839da1c2c9b66e9fa2a45f33a5661R138

The cli still needs to prepare the backend-specific options struct, and also cannot import the entbackend options struct on 32-bit (as that would pull in the whole package).

It's a bit of a false encapsulation as the options structs need to be known, I'm not sure it is better than how it was previously.

Comment (lumjjb) [reply cmd/guacgql/cmd/server.go:64]: Ah gotcha, i think that's good for now then. I do have an idea that could maybe help a bit but not blocking.

**Pull request merged**